src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ExceptionInfoLookupTableNode.cs

```diff
@@ -0,0 +1,153 @@
+        }
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            return new ObjectData(_ehInfoBuilder.ToArray(), Array.Empty<Relocation>(), alignment: 1, definedSymbols: new ISymbolDefinitionNode[] { this });
+        }
+
+        protected override string GetName(NodeFactory context)
+        {
+            Utf8StringBuilder sb = new Utf8StringBuilder();
+            AppendMangledName(context.NameMangler, sb);
+            return sb.ToString();
+        }
+    }
+
+    public class ExceptionInfoLookupTableNode : HeaderTableNode
+    {
+        private List<MethodWithGCInfo> _methodNodes;
+        private List<int> _ehInfoOffsets;
+
+        private readonly NodeFactory _nodeFactory;
+
+        private readonly EHInfoNode _ehInfoNode;
+
+        public ExceptionInfoLookupTableNode(NodeFactory nodeFactory)
+            : base(nodeFactory.Target)
+        {
+            _nodeFactory = nodeFactory;
+            _ehInfoNode = new EHInfoNode();
+        }
+
+        public override void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(nameMangler.CompilationUnitPrefix);
+            sb.Append("__ReadyToRunExceptionInfoLookupTable@");
+            sb.Append(Offset.ToString());
+        }
+
+        internal void LayoutMethodsWithEHInfo()
+        {
+            if (_methodNodes != null)
+            {
+                // Already initialized
+                return;
+            }
+
+            _methodNodes = new List<MethodWithGCInfo>();
+            _ehInfoOffsets = new List<int>();
+
+            foreach (MethodDesc method in _nodeFactory.MetadataManager.GetCompiledMethods())
+            {
+                MethodWithGCInfo methodCodeNode = _nodeFactory.MethodEntrypoint(method) as MethodWithGCInfo;
+                if (methodCodeNode == null)
+                {
+                    methodCodeNode = ((ExternalMethodImport)_nodeFactory.MethodEntrypoint(method))?.MethodCodeNode;
+                    if (methodCodeNode == null)
+                        continue;
+                }
+
+                ObjectData ehInfo = methodCodeNode.EHInfo;
+                if (ehInfo != null && ehInfo.Data.Length != 0)
+                {
+                    _methodNodes.Add(methodCodeNode);
+                    _ehInfoOffsets.Add(_ehInfoNode.AddEHInfo(ehInfo.Data));
+                }
+            }
+        }
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            if (relocsOnly)
+            {
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolDefinitionNode[] { this });
+            }
+
+            if (_methodNodes == null)
+                LayoutMethodsWithEHInfo();
+
+            ObjectDataBuilder exceptionInfoLookupBuilder = new ObjectDataBuilder(factory, relocsOnly);
+            exceptionInfoLookupBuilder.RequireInitialAlignment(8);
```

Comment:
Just to double check, 8? or pointer size?

Reply:
I'm actually not sure where I got that from, perhaps just from the fact that the entries in the table have 8 bytes (2 RVA's). For now I have changed the constant to "2 * sizeof(uint)" to make it easier to see.